pkg/middleware/request_tracing.go

```diff
@@ -3,20 +3,34 @@ package middleware
 import (
 	"fmt"
 	"net/http"
+	"strings"
 
 	opentracing "github.com/opentracing/opentracing-go"
 	"github.com/opentracing/opentracing-go/ext"
 
 	"gopkg.in/macaron.v1"
 )
 
-func RequestTracing(handler string) macaron.Handler {
+const routeOperationNameKey = "route.operation.name"
+
+func ProvideRouteOperationName(handler string) macaron.Handler {
+	return func(res http.ResponseWriter, req *http.Request, c *macaron.Context) {
+		c.Data[routeOperationNameKey] = handler
+	}
+}
+
+func RequestTracing() macaron.Handler {
 	return func(res http.ResponseWriter, req *http.Request, c *macaron.Context) {
+		if !strings.HasPrefix(c.Req.URL.Path, "/api/") {
```

Comment:
I think we want traces for all endpoints except /metrics and healthcheck.

Reply:
I did think of that actually, but was unsure what enabling this for everything would mean. Even static resources/requests like css, javascript, html?

Reply:
No, I guess static assets won't be helpful but it would be interesting for all regular requests. I guess any route that's defined in https://github.com/grafana/grafana/blob/master/pkg/api/api.go#L44 would be interesting.

Reply:
So if the value is added to the context we can assume that its interesting :)

Reply:
Good point